packages/amplication-prisma-db/prisma/schema.prisma

```diff
@@ -127,39 +127,54 @@ model User {
 }
 
 model Role {
-  id          String    @id @default(cuid())
-  createdAt   DateTime  @default(now())
-  updatedAt   DateTime  @updatedAt
-  name        String
-  key         String
-  deletedAt   DateTime?
-  description String?
-  permissions String[]
-  workspaceId String
-  workspace   Workspace @relation(fields: [workspaceId], references: [id], onDelete: Cascade, onUpdate: Cascade)
-  teams       Team[]
+  id              String           @id @default(cuid())
+  createdAt       DateTime         @default(now())
+  updatedAt       DateTime         @updatedAt
+  name            String
+  key             String
+  deletedAt       DateTime?
+  description     String?
+  permissions     String[]
+  workspaceId     String
+  workspace       Workspace        @relation(fields: [workspaceId], references: [id], onDelete: Cascade, onUpdate: Cascade)
+  teams           Team[]
+  teamAssignments TeamAssignment[]
 
   @@unique([workspaceId, key], map: "Role.workspaceId_key_unique")
   @@unique([workspaceId, name], map: "Role.workspaceId_name_unique")
 }
 
 model Team {
-  id          String      @id @default(cuid())
-  createdAt   DateTime    @default(now())
-  updatedAt   DateTime    @updatedAt
-  name        String
-  description String?
-  members     User[]
-  workspaceId String
-  workspace   Workspace   @relation(fields: [workspaceId], references: [id], onDelete: Cascade, onUpdate: Cascade)
-  deletedAt   DateTime?
-  ownerships  Ownership[]
-  color       String?
-  roles       Role[]
+  id             String           @id @default(cuid())
+  createdAt      DateTime         @default(now())
+  updatedAt      DateTime         @updatedAt
+  name           String
+  description    String?
+  members        User[]
+  workspaceId    String
+  workspace      Workspace        @relation(fields: [workspaceId], references: [id], onDelete: Cascade, onUpdate: Cascade)
+  deletedAt      DateTime?
+  ownerships     Ownership[]
+  color          String?
+  roles          Role[]
+  teamAssignment TeamAssignment[]
 
   @@unique([workspaceId, name], map: "Team.workspaceId_name_unique")
 }
 
+model TeamAssignment {
+  id         String   @id @default(cuid())
+  createdAt  DateTime @default(now())
+  updatedAt  DateTime @updatedAt
+  resourceId String
+  resource   Resource @relation(fields: [resourceId], references: [id], onDelete: Cascade)
+  teamId     String
+  team       Team     @relation(fields: [teamId], references: [id], onDelete: Cascade)
+  roles      Role[]
```

Comment:
a bit weird to me why we have roles on team assignments and team assignments on roles. i guess that prisma's way of allowing you to have this bidirectional access?

working in django (python), they have this same concept but you only need to put the field once and you can access from both sides.